prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py

```diff
@@ -0,0 +1,206 @@
+    """Load Vehicle Miles Traveled (VMY) per capita in urban areas
+
+    :param str path: path to Department of Transportation's Transportation Health Tools
+        (local or url)
+    :return: (*tuple*) -- Series. Indices are state abbreviations and values are
+        population in ua (first element) or state (second element)
+    """
+    df_ua = pd.read_excel(
+        path,
+        sheet_name="Urbanized Area",
+        index_col=0,
+        usecols=[0, 3],
+        names=["UA", "VMT per Capita (daily)"],
+    )
+
+    df_state = pd.read_excel(
+        path,
+        sheet_name="State",
+        index_col=0,
+        usecols=[0, 39],
+        names=["State", "VMT per Capita (annual)"],
+    )
+
+    return (
+        df_ua.squeeze().loc[lambda x: x != "[no data]"],
+        df_state.rename(index=state2abv).squeeze(),
+    )
+
+
+def calculate_vmt_for_ua(census_ua, tht_ua):
+    """Calculate the total annual Vehicle Miles Traveled (VMT) in urban areas
+
+    :param dict census_ua: dictionary as returned by :func:`load_census_ua`
+    :param pandas.Series tht_ua: vmt per capita in urban areas as returned by
+        :func:`load_dot_vmt_per_capita`
+    :return: (*dict*) -- keys are state abbreviation and values are data frames giving
+        annual vmt by urban areas.
+    """
+
+    tht_ua_format = tht_ua.copy()
+    vmt_for_ua = {}
+
+    pattern = [",", "-", ".", " "]
+    for p in pattern:
+        tht_ua_format.index = tht_ua_format.index.str.replace(p, "", regex=True)
+    format2original = dict(zip(tht_ua_format.index, tht_ua.index))
+
+    for s in census_ua:
+        census_ua_format = census_ua[s].copy()
+        for p in pattern:
+            census_ua_format.index = census_ua_format.index.str.replace(
+                p, "", regex=True
+            )
+        common = set(tht_ua_format.index).intersection(set(census_ua_format.index))
+        vmt_for_ua[s] = (
+            pd.DataFrame(
+                {
+                    "Annual VMT": [
+                        365 * tht_ua_format.loc[i] * census_ua_format.loc[i]
+                        for i in common
+                    ]
+                },
+                index=list(common),
+            )
+            .rename(index=format2original)
+            .squeeze()
+        )
+
+    return vmt_for_ua
+
+
+def calculate_vmt_for_state(census_state, tht_state):
+    """Calculate the total annual Vehicle Miles Traveled (VMT) in states
+
+    :param dict census_state: dictionary as returned by :func:`load_census_state`
+    :param pandas.Series tht_state: vmt per capita in states as returned by
+        :func:`load_dot_vmt_per_capita`
+    :return: (*pandas.Series*) -- Indices are state abbreviations and values are and
+        values are annual vmt in state.
+    """
+    common = set(tht_state.index).intersection(set(census_state.index))
+
+    vmt_for_state = pd.Series(
+        [tht_state.loc[i] * census_state.loc[i] for i in common], index=list(common)
+    )
```

Comment:
Does it equivalent to `vmt_for_state = tht_state.loc[common]*census_state.loc[common]`?

Reply:
Yep

Reply:
Changed